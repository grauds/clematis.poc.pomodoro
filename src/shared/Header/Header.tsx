import * as React from 'react';
import { hot } from 'react-hot-loader/root';
import { Title } from './Title';

import styles from './header.less';
import { Stats } from './Stats';

function HeaderComponent(): React.JSX.Element {
  return (
    <header className={styles.header}>
      <Title />
      <Stats />
    </header>
  );
}

export const Header = hot(HeaderComponent);
