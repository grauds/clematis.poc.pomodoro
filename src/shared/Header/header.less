.header {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  padding: 30px 51px 30px 51px;
  -webkit-box-shadow: 0px 7px 11px 0px rgba(34, 60, 80, 0.1);
  -moz-box-shadow: 0px 7px 11px 0px rgba(34, 60, 80, 0.1);
  box-shadow: 0px 7px 11px 0px rgba(34, 60, 80, 0.1);
}
