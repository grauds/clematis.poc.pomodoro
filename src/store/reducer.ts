import { ActionCreator, AnyAction, Reducer } from "redux";
import { Days, IDay, ITask, IWeek, Weeks } from "../types/model";

export type RootState = {
  tasks: ITask[];
  day: IDay;
  week: IWeek;
};

const initialState: RootState = {
  tasks: [],
  day: Days[0],
  week: Weeks[0],
};

const ADD_TASK = "ADD_TASK";
const REMOVE_TASK = "REMOVE_TASK";
const SET_CURRENT_DAY = "SET_CURRENT_DAY";
const SET_CURRENT_WEEK = "SET_CURRENT_WEEK";

export const addTask: ActionCreator<AnyAction> = (task: ITask) => ({
  type: ADD_TASK,
  task,
});

export const removeTask: ActionCreator<AnyAction> = (task: ITask) => ({
  type: REMOVE_TASK,
  task,
});

export const setCurrentDay: ActionCreator<AnyAction> = (day: IDay) => ({
  type: SET_CURRENT_DAY,
  day,
});

export const setCurrentWeek: ActionCreator<AnyAction> = (week: IWeek) => ({
  type: SET_CURRENT_WEEK,
  week,
});

export const rootReducer: Reducer<RootState> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case ADD_TASK:
      return {
        ...state,
        tasks: state.tasks.concat(action.task),
      };
    case REMOVE_TASK:
      return {
        ...state,
        tasks:
          state.tasks.indexOf(action.task) !== -1
            ? state.tasks.splice(action.task, 1)
            : state.tasks,
      };
    case SET_CURRENT_DAY:
      return {
        ...state,
        day: action.day,
      };  
    case SET_CURRENT_WEEK:
      return {
        ...state,
        week: action.week,
      };
    default:
      return state;
  }
};
