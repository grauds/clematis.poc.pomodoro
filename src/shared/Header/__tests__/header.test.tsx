import React from 'react';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import { initialState, rootReducer } from '@/store/reducer';
import { createStore } from 'redux';

import { Header } from '../Header';

import { render } from '@testing-library/react';

const store = createStore(rootReducer, initialState);
describe('Header', () => {
  test('should render', () => {
    const {container}= render(
      <Provider store={store}>
        <MemoryRouter>
          <Header />
        </MemoryRouter>
      </Provider>,
    );
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
