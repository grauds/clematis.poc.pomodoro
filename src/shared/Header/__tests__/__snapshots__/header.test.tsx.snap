// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render 1`] = `
<div>
  <header
    class="header"
  >
    <a
      class="title"
      href="/"
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_23799_226)"
        >
          <path
            d="M38.9151 23.2834C38.9151 33.7058 30.466 40 20.0437 40C9.62098 40 1.17188 31.5509 1.17188 21.1282C1.17188 10.7059 9.88496 4.2981 20.3073 4.2981C30.73 4.2981 38.9151 12.8607 38.9151 23.2834Z"
            fill="#DC3E22"
          />
          <path
            d="M28.238 12.6066C27.3211 11.673 25.8377 10.8048 24.733 10.551C25.3401 10.0127 25.4623 9.99494 26.2227 9.61816C28.1713 8.65365 31.0576 8.56485 31.0576 8.56485C31.0576 8.56485 27.6509 6.8042 25.1601 6.91468C24.5259 6.94257 23.8571 7.16658 23.2118 7.48403C23.5757 6.97054 23.9205 6.45998 24.1409 6.07643C24.8152 4.90368 25.524 3.42627 25.524 3.42627C25.524 3.42627 22.9122 3.56573 21.7008 5.01565C21.2407 5.56645 20.8934 6.26625 20.6392 6.92275C20.1878 6.40419 19.6896 5.94242 19.1913 5.58195C16.6999 3.77896 12.7192 4.16903 12.7192 4.16903C12.7192 4.16903 15.7263 5.87486 17.0793 7.57656C17.6076 8.2411 18.1437 8.54842 18.4642 9.29352C17.3564 9.05367 14.8569 9.13565 13.63 9.59057C10.4771 10.7599 9.11852 15.4649 9.11852 15.4649C9.11852 15.4649 12.1952 13.3443 16.3813 11.8565C17.3017 11.5295 18.2748 11.4429 19.1229 11.4578C18.7379 12.0575 18.3173 12.8363 17.999 13.7546C17.2247 15.9904 18.2479 21.3113 18.2479 21.3113C18.2479 21.3113 20.4896 18.1647 21.403 15.6157C21.8718 14.3073 21.9879 12.9936 21.9904 12.0242C22.8217 12.3931 23.8009 12.9319 24.5326 13.398C28.2794 15.7852 30.072 20.1435 30.072 20.1435C30.072 20.1435 30.5941 15.006 28.238 12.6066Z"
            fill="#899441"
          />
          <path
            d="M20.5008 10.3094C20.4889 10.3094 20.477 10.3091 20.4651 10.3088C19.7242 10.2896 19.1391 9.67376 19.1572 8.9334C19.1587 8.86931 19.2234 4.36125 16.7191 2.40111C16.135 1.94395 16.0318 1.09984 16.489 0.515424C16.9465 -0.0686834 17.7906 -0.171833 18.3747 0.285626C21.9559 3.08806 21.8491 8.76128 21.843 9.00145C21.8238 9.73083 21.2262 10.3094 20.5008 10.3094Z"
            fill="#A8B64F"
          />
        </g>
        <defs>
          <clippath
            id="clip0_23799_226"
          >
            <rect
              fill="white"
              height="40"
              width="40"
            />
          </clippath>
        </defs>
      </svg>
       
      <div
        class="brand"
      >
        pomodoro_box
      </div>
    </a>
    <div
      class="container"
    >
      <a
        class="stats"
        href="/statistics"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_23799_232)"
          >
            <path
              d="M10 20H14V4H10V20ZM4 20H8V12H4V20ZM16 9V20H20V9H16Z"
              fill="#DC3E22"
            />
          </g>
          <defs>
            <clippath
              id="clip0_23799_232"
            >
              <rect
                fill="white"
                height="24"
                width="24"
              />
            </clippath>
          </defs>
        </svg>
         
        <div
          class="text"
        >
          Статистика
        </div>
      </a>
      <button
        class="settings"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.1915 10.8165C16.2249 10.5499 16.2499 10.2832 16.2499 9.99988C16.2499 9.71654 16.2249 9.44988 16.1915 9.18321L17.9499 7.80821C18.1082 7.68321 18.1499 7.45821 18.0499 7.27488L16.3832 4.39154C16.3341 4.30556 16.2559 4.23989 16.1628 4.2063C16.0696 4.1727 15.9676 4.17338 15.8749 4.20821L13.7999 5.04154C13.3665 4.70821 12.8999 4.43321 12.3915 4.22488L12.0749 2.01654C12.0611 1.91849 12.012 1.82884 11.9368 1.76441C11.8616 1.69997 11.7655 1.66517 11.6665 1.66654H8.3332C8.12487 1.66654 7.94987 1.81654 7.92487 2.01654L7.6082 4.22488C7.09987 4.43321 6.6332 4.71654 6.19987 5.04154L4.12487 4.20821C4.07655 4.19182 4.02589 4.18337 3.97487 4.18321C3.8332 4.18321 3.69153 4.25821 3.61653 4.39154L1.94987 7.27488C1.84153 7.45821 1.89153 7.68321 2.04987 7.80821L3.8082 9.18321C3.77487 9.44988 3.74987 9.72488 3.74987 9.99988C3.74987 10.2749 3.77487 10.5499 3.8082 10.8165L2.04987 12.1915C1.89153 12.3165 1.84987 12.5415 1.94987 12.7249L3.61653 15.6082C3.66566 15.6942 3.7438 15.7599 3.83695 15.7935C3.9301 15.8271 4.03217 15.8264 4.12487 15.7915L6.19987 14.9582C6.6332 15.2915 7.09987 15.5665 7.6082 15.7749L7.92487 17.9832C7.94987 18.1832 8.12487 18.3332 8.3332 18.3332H11.6665C11.8749 18.3332 12.0499 18.1832 12.0749 17.9832L12.3915 15.7749C12.8999 15.5665 13.3665 15.2832 13.7999 14.9582L15.8749 15.7915C15.9249 15.8082 15.9749 15.8165 16.0249 15.8165C16.1665 15.8165 16.3082 15.7415 16.3832 15.6082L18.0499 12.7249C18.1499 12.5415 18.1082 12.3165 17.9499 12.1915L16.1915 10.8165V10.8165ZM14.5415 9.39154C14.5749 9.64988 14.5832 9.82488 14.5832 9.99988C14.5832 10.1749 14.5665 10.3582 14.5415 10.6082L14.4249 11.5499L15.1665 12.1332L16.0665 12.8332L15.4832 13.8415L14.4249 13.4165L13.5582 13.0665L12.8082 13.6332C12.4499 13.8999 12.1082 14.0999 11.7665 14.2415L10.8832 14.5999L10.7499 15.5415L10.5832 16.6665H9.41653L9.2582 15.5415L9.12487 14.5999L8.24153 14.2415C7.8832 14.0915 7.54987 13.8999 7.21653 13.6499L6.4582 13.0665L5.57487 13.4249L4.51653 13.8499L3.9332 12.8415L4.8332 12.1415L5.57487 11.5582L5.4582 10.6165C5.4332 10.3582 5.41653 10.1665 5.41653 9.99988C5.41653 9.83321 5.4332 9.64154 5.4582 9.39154L5.57487 8.44988L4.8332 7.86654L3.9332 7.16654L4.51653 6.15821L5.57487 6.58321L6.44153 6.93321L7.19153 6.36654C7.54987 6.09988 7.89153 5.89988 8.2332 5.75821L9.11653 5.39988L9.24987 4.45821L9.41653 3.33321H10.5749L10.7332 4.45821L10.8665 5.39988L11.7499 5.75821C12.1082 5.90821 12.4415 6.09988 12.7749 6.34988L13.5332 6.93321L14.4165 6.57488L15.4749 6.14988L16.0582 7.15821L15.1665 7.86654L14.4249 8.44988L14.5415 9.39154ZM9.99987 6.66654C8.1582 6.66654 6.66653 8.15821 6.66653 9.99988C6.66653 11.8415 8.1582 13.3332 9.99987 13.3332C11.8415 13.3332 13.3332 11.8415 13.3332 9.99988C13.3332 8.15821 11.8415 6.66654 9.99987 6.66654ZM9.99987 11.6665C9.0832 11.6665 8.3332 10.9165 8.3332 9.99988C8.3332 9.08321 9.0832 8.33321 9.99987 8.33321C10.9165 8.33321 11.6665 9.08321 11.6665 9.99988C11.6665 10.9165 10.9165 11.6665 9.99987 11.6665Z"
            fill="#DC3E22"
          />
        </svg>
         
        <div
          class="text"
        >
          Настройки
        </div>
      </button>
    </div>
  </header>
</div>
`;
